stix2/datastore/__init__.py

```diff
@@ -24,15 +24,35 @@ def make_id():
     return str(uuid.uuid4())
 
 
+class DataSourceError(Exception):
+    """General DataSource error instance, used primarily for wrapping
+    lower level errors
+
+    Args:
+        message (str): error message
+        root_exception (Exception): Exception instance of root exception
+            in the case that DataSourceError is wrapping a lower level or
+            other exception
+    """
+    def __init__(self, message, root_exception=None):
+        self.message = message
+        self.root_exception = root_exception
+
+    def __str__(self):
+        if self.root_exception:
+            self.message = "{} \"{}\"".format(self.message, self.root_exception)
```

Comment:
This is dangerous because if you call it multiple times, the message will keep getting longer.